middlewares/validators/userStatus.js

```diff
@@ -0,0 +1,72 @@
+const joi = require("joi");
+const { userState } = require("../../constants/userStatus");
+
+const validateUserStatus = async (req, res, next) => {
+  const schema = joi
+    .object({
+      userId: joi.string().trim().required(),
+      currentStatus: joi.object().keys({
+        state: joi.string().trim().valid(userState.IDLE, userState.ACTIVE, userState.OOO),
+        updatedAt: joi.number().required().strict(),
+        from: joi.number().required().strict(),
+        until: joi
+          .any()
+          .when("state", { is: userState.OOO, then: joi.number().required().strict(), otherwise: joi.optional() }),
```

Comment:
Would until not be number | null ?
Do we have better way than any?

Reply:
for updating User Status `until` is an optional field in the request body. It may or may not be present. Its required only for `OOO` status. Hence only in the case of `OOO`, I am checking if it should be a number. And In the controller, we have set this to an empty string if the state is not `OOO`.

Reply:
Since anyway the value of `until` is going to be overwritten with an empty string for everything apart from `OOO` so kept it joi.any().